src/input/mod.rs

```diff
@@ -0,0 +1,490 @@
+use crate::pyo3_extensions::{TdPyAny, TdPyIterator};
+
+use log::info;
+use pyo3::basic::CompareOp;
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::types::PyString;
+use rdkafka::client::ClientContext;
+use rdkafka::config::{ClientConfig, RDKafkaLogLevel};
+use rdkafka::consumer::base_consumer::BaseConsumer;
+use rdkafka::consumer::{Consumer, ConsumerContext, Rebalance};
+use rdkafka::error::KafkaResult;
+use rdkafka::message::Message;
+use rdkafka::topic_partition_list::TopicPartitionList;
+use std::time::Duration;
+use timely::dataflow::InputHandle;
+use tokio::runtime::Runtime;
+
+/// Advance to the supplied epoch.
+///
+/// When providing input to a Dataflow, work cannot complete until
+/// there is no more data for a given epoch.
+///
+/// AdvanceTo is the signal to a Dataflow that the frontier has moved
+/// beyond the current epoch, and that items with an epoch less than
+/// the epoch in AdvanceTo can be worked to completion.
+///
+/// Using AdvanceTo and Emit is only necessary when using `spawn_cluster`
+/// and `cluster_main()` as `run()` and `run_cluster()` will yield AdvanceTo
+/// and Emit for you. Likewise, they are only required when using a
+/// manual input configuration.
+///
+///
+/// See also: `inputs.yield_epochs()`
+///
+/// >>> def input_builder(worker_index, worker_count, resume_epoch):
+/// ...     for i in range(10):
+/// ...         yield AdvanceTo(i) # Advances the epoch to i
+/// ...         yield Emit(i) # Adds the input i at epoch i
+#[pyclass(module = "bytewax")]
```

Comment:
Might be out of scope for this PR, but maybe think about moving `bytewax.Emit` into `bytewax.inputs.Emit`. In that case, these Python module names will need to be changed.